core/src/main/java/org/mskcc/cbio/portal/servlet/QueryBuilder.java

```diff
@@ -243,79 +244,130 @@ protected void doPost(HttpServletRequest httpServletRequest,
+                            }
+                            if(mutationProfile != null) {
+                                geneticProfileIdSet.add(mutationProfile.getStableId());
+                            }
+                            httpServletRequest.setAttribute(GENETIC_PROFILE_IDS, geneticProfileIdSet);
                         }
+                    } 
+                    //multiple and virtual study
+                    else {
+                        _isVirtualStudy = true;
+                        if (cancerStudyId != null && !cancerStudyId.equals("all")) {
+                            cancerStudyList = null;
+                        } else {
+                            if ((cancerStudyId == null || cancerStudyId.equals("all")) 
+                            && (cancerStudyList!= null && !cancerStudyList.equals("all"))) {
+                                cancerStudyId = "all";
+                            }
+                        }
+                        
+                    }
+                    httpServletRequest.setAttribute(IS_VIRTUAL_STUDY, _isVirtualStudy);
+                    if(httpServletRequest.getParameter(CASE_SET_ID) != null) {
+                    	    httpServletRequest.setAttribute(CASE_SET_ID, httpServletRequest.getParameter(CASE_SET_ID));
+                    	    if (httpServletRequest.getParameter(CASE_SET_ID).equals("-1")) {
+                                httpServletRequest
+                                    .setAttribute(CASE_IDS,httpServletRequest
+                                                                .getParameter(CASE_IDS)
+                                                                .replaceAll("\\\\n", "\n")
+                                                                .replaceAll("\\\\t", "\t"));
+                            }
+                    	
                     }
-	            }
+                }
             }
             httpServletRequest.setAttribute(CANCER_STUDY_ID, cancerStudyId);
             httpServletRequest.setAttribute(CANCER_STUDY_LIST, cancerStudyList);
-			
+            
             if(cancerStudyId != null) {
-            		CohortDetails cohortDetails;
+                String queriedStudies = cancerStudyId;
                 if (cancerStudyId.equals("all") 
-                		&& cancerStudyList != null) { // multiple studies
-                    cohortDetails = new CohortDetails(cancerStudyList.split(","), _isVirtualStudy);
-                } else { // single study (can be VC here)
-                    cohortDetails = new CohortDetails(new String[]{cancerStudyId}, _isVirtualStudy);
+                    && cancerStudyList != null) {
+                    queriedStudies = cancerStudyList;
+                }
+                    
+                Set<String> inputCohortIds = new HashSet<>(Arrays.asList(queriedStudies.split(",")));
+                Set<VirtualStudy> studies = getStudies(inputCohortIds);
```

Comment:
getProcessedStudyMap in CancerStudyView.java and getStudies in this file are almost identical, ever considered to consolidate them together?

Reply:
Yes, that true. Thought of not cleaning it now as most of these would be gone once we have single page application. Or I could do all the cleanup in another PR

Reply:
OK, we could leave it for now. Just want to have the conversation started.